lib/prime.rb

```diff
@@ -132,7 +209,8 @@ def method_added(method) # :nodoc:
   #   Upper bound of prime numbers. The iterator stops after it
   #   yields all prime numbers p <= +ubound+.
   #
-  def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  #def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  def each(ubound = nil, generator = NextPrimeGenerator.new, &block)
```

Comment:
Yes, I like the simplification at the end. 

I used the array because it's visually simpler to see what the test is, but it can be replaced with:
```
      return false unless [1, 5].include?(self % 6)
      return false unless 6.gcd(self % 6) == 1
```
6 = 2*3 is the P3 Prime Generator (PG) modulus, so for any number > 3, the residue ` n  mod 6` has to be either 1 or 5 (coprimes to 6), which is equivalent to directly testing if the residue of  `n mod 6` is coprime to 6.

The math for all of this is explained in my Prime-Utils Handbook.